tensorrt_llm/_torch/modules/attention.py

```diff
@@ -126,24 +157,48 @@ def forward(
         attention_mask: PredefinedAttentionMask = PredefinedAttentionMask.
         CAUSAL,
         mrope_config: Optional[dict] = None,
+        all_reduce_params: Optional[AllReduceParams] = None,
         **kwargs,
     ) -> torch.Tensor:
         qkv = self.qkv_proj(hidden_states)
         is_fused_qkv = False
         if isinstance(self.attn, TrtllmAttention):
             is_fused_qkv = True
 
+        if self.qk_norm is not None:
+            # TODO: make this more efficient.
+            q, k, v = qkv.split([self.q_size, self.kv_size, self.kv_size],
+                                dim=-1)
+            do_multi_stream = torch.cuda.is_current_stream_capturing(
+            ) and self.aux_stream is not None
+            if do_multi_stream:
+                self.ln_events[0].record()
+                k = self.qk_norm(k)
+                with torch.cuda.stream(self.aux_stream):
+                    self.ln_events[0].wait()
+                    q = self.qk_norm(q)
+                    self.ln_events[1].record()
+                self.ln_events[1].wait()
+            else:
+                q = self.qk_norm(q)
+                k = self.qk_norm(k)
+            qkv = torch.concat(
+                [q.contiguous(), k.contiguous(),
+                 v.contiguous()], dim=-1)
```

Comment:
```suggestion
            qkv = torch.concat(
                [q, k,
                 v], dim=-1)
```
concat supports strided inputs